### PR TITLE
Android: Provide dev/build environment using Vagrant

### LINKED ISSUE #1829: Provide Android dev/build environment
BOINC should provide an easy to use build environment for developing and building Android artefacts like the client and manager.

### PR DESCRIPTION
**Pull request by brevilo**
This resolves #1829

### CODE REVIEW

**Comment by brevilo** [comment]
FYI, just added a refined version of the README...

File: android/Vagrant.README.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+      1. `./build_all.sh`
+   1. Start Android Studio
+   1. No need to change anything in the setup assitant (just complete it)
+      * OK / Next / Next / Finish / Finish
+   1. Import the BOINC App as *Gradle* project from: `~/BOINC/android/BOINC`
+   1. Ignore potential Gradle Plugin warning: *Don't remind me again*
+1. Hook up your Android device via USB (and remember to attach it to VirtualBox)
+1. Happy hacking :-)
+
+## Know limitations
```
Comment (ChristianBeer) [inline android/Vagrant.README.md]: minor typo; it should be "Known limitations"

**Comment by brevilo** [comment]
> I had to apply #1830 to get the arm build working

Well, of course 😁

File: android/Vagrant.README.md (hunk 1 above)
Comment (brevilo) [inline android/Vagrant.README.md]: Fixed

**Comment by davidpanderson** [comment]
Thanks!

**Pull request merged**